### PR TITLE
Added fallback when port in use

### PR DESCRIPTION
**Pull request by Bearbobs**

### CODE REVIEW

File: packages/cli/src/Server.ts [hunk 1]
```diff
@@ -2898,6 +2898,16 @@ export async function start(): Promise<void> {
 
 		void InternalHooksManager.getInstance().onServerStarted(diagnosticInfo);
 	});
+
+	// @ts-ignore
+	server.on('error', (error) => {
```
Comment (ivov) [inline packages/cli/src/Server.ts:2903]:
```suggestion
	server.on('error', (error: Error & { code: string }) => {
```

File: packages/cli/src/Server.ts [hunk 2]
```diff
@@ -2898,6 +2898,16 @@ export async function start(): Promise<void> {
...
+		if (error.code === 'EADDRINUSE') {
+			console.log(
+				`n8n's port ${PORT} is already in use. Do you have another instance of n8n running already?`,
+			);
+			process.exit(0);
```
Comment (ivov) [inline packages/cli/src/Server.ts:2908]:
```suggestion
			process.exit(1);
```

Comment (Bearbobs) [reply packages/cli/src/Server.ts]: Other cli requirements also exit in code `0` ( https://github.com/n8n-io/n8n/blob/6453996de30e9635bda373f9f120a54621a8a69f/packages/cli/bin/n8n#L29)
Should we update them as well?

Comment (ivov) [reply packages/cli/src/Server.ts]: Afaik non-zero exit signifies failure. Cannot say why the other checks don't use it.

Comment (Bearbobs) [reply packages/cli/src/Server.ts]: as discussed with @krynble , updating other's as well

**Comment by krynble** [comment]
Nice, the error message is displayed correctly but we still get an "ugly" output in the terminal.

Do you think we can capture the error and suppress it?

Screenshot from 2021-12-06 17-51-23

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Thanks a lot. Got merged.

**Comment by janober** [comment]
Got released with `n8n@0.153.0`

**Pull request merged**